contributors/design-proposals/cloudprovider-storage-metrics.md

```diff
@@ -0,0 +1,65 @@
+# Cloud Provider (specifically GCE and AWS) metrics for Storage API calls
+
+## Goal
+
+Kubernetes should provide metrics such as - count & latency percentiles
+for cloud provider API it uses to provision persistent volumes.
+
+In a ideal world - we would want these metrics for all cloud providers
+and for all API calls kubernetes makes but to limit the scope of this feature
+we will implement metrics for:
+
+* GCE
+* AWS
+
+We will also implement metrics only for storage API calls for now. This feature
+does introduces hooks into kubernetes code which can be used to add additonal metrics
+but we only focus on storage API calls here.
+
+## Motivation
+
+* Cluster admins should be able to monitor Cloud API usage of Kubernetes. It will help
+  them detect problems in certain scenarios which can blow up the API quota of Cloud
+  provider.
+* Cluster admins should also be able to monitor health and latency of Cloud API on
+  which kubernetes depends on.
+
+## Implementation
+
+### Metric format and collection
+
+Metrics emitted from cloud provider will fall under category of service metrics
+as defined in [Kubernetes Monitoring Architecture](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/monitoring_architecture.md).
+
+
+The metrics will be emitted using [Prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/) and available for collection
+from `/metrics` HTTP endpoint of kubelet, controller etc. All Kubernetes core components already emit
+metrics on `/metrics` HTTP endpoint. This proposal merely extends available metrics to include Cloud provider metrics as well.
+
+
+Any collector which can parse Prometheus metric format should be able to collect
+metrics from these endpoints.
+
+A more detailed description of monitoring pipeline can be found here - https://github.com/kubernetes/community/blob/master/contributors/design-proposals/monitoring_architecture.md#monitoring-pipeline
+
+
+#### Metric Types
+
+Since we are mostly interested in count of API calls and its latency - [Summary](https://prometheus.io/docs/concepts/metric_types/#summary) metric type should do the job.
+Summary type will enable us to compute percentiles and count easily. We do take a hit
+in accuracy when it comes to percentile computation but I think we should be fine with that.
+
+We will be using `SummaryVec` type so as we can attach dimensions at runtime. Whenever available
+`namespace` will reported as a dimension with the metric.
+
+### GCE Implementation
```

Comment:
Could you be more specific in the implementations for both AWS and GCE. e.g. "We will instrument the following API calls for latency, success/failures..."